cobalt/browser/cobalt_content_browser_client.cc

```
@@ -295,4 +327,118 @@ bool CobaltContentBrowserClient::WillCreateURLLoaderFactory(
+    // intended to be a temporary rollout stage, and the short timeout causes
+    // flakiness when the test server takes just a tad too long to respond.
+    feature_overrides.emplace_back(
+        std::cref(
+            network::features::kPrivateNetworkAccessPreflightShortTimeout),
+        base::FeatureList::OVERRIDE_DISABLE_FEATURE);
+  }
+
+  feature_list->InitializeFromCommandLine(
+      command_line.GetSwitchValueASCII(::switches::kEnableFeatures),
+      command_line.GetSwitchValueASCII(::switches::kDisableFeatures));
+
+  // This needs to happen here: After the InitFromCommandLine() call,
+  // because the explicit cmdline --disable-features and --enable-features
+  // should take precedence over these extra overrides. Before the call to
+  // SetInstance(), because overrides cannot be registered after the FeatureList
+  // instance is set.
+  feature_list->RegisterExtraFeatureOverrides(feature_overrides);
+
+  // Add Cobalt custom code to create field trial here.
+  base::FieldTrial* cobalt_field_trial = base::FieldTrialList::CreateFieldTrial(
+      kCobaltExperimentName, kCobaltGroupName);
+  CHECK(cobalt_field_trial) << "Unexpected name conflict.";
+
+  const base::Value::Dict& feature_map =
+      exp_config_->GetDict(kExperimentConfigFeature);
+  const base::Value::Dict& param_map =
+      exp_config_->GetDict(kExperimentConfigFeatureParams);
+
+  for (const auto kvpair : feature_map) {
+    if (kvpair.second.is_bool() && kvpair.second.GetBool()) {
+      feature_list->RegisterFieldTrialOverride(
+          kvpair.first,
+          base::FeatureList::OverrideState::OVERRIDE_ENABLE_FEATURE,
+          cobalt_field_trial);
+    } else if (kvpair.second.is_bool()) {
+      feature_list->RegisterFieldTrialOverride(
+          kvpair.first,
+          base::FeatureList::OverrideState::OVERRIDE_DISABLE_FEATURE,
+          cobalt_field_trial);
+    } else {
+      // TODO(b/407734134): Register UMA here for non boolean feature value.
```

And throw a fatal error

---

Same as above, the param would fallback to default value without a fatal error.

---

As an alternative, could we add a histogram along with a log? Something to track how often we are getting malformed experiment configs?

---

Histograms are definitely on the roadmap in b/407734134. I'd personally be fine punting for now and keeping the TODO that's here.

---

> Same as above, the param would fallback to default value without a fatal error.

I'm worried an error is not enough. If you look at the stream of output from Cobalt, there's already many errors. Additionally, a Kabuki engineer would never know if they introduced a bad type. 

@jellefoks I wonder if Jelle has input here. We're trying to decide if Kabuki passes a bad parameter (e.g., a non-bool for something that should be bool), if it's reasonable to crash instead of just log an error. Another alternative is a UMA histogram, but we'd need to make sure we're monitoring it very closely and we don't really have a lot of precedent for that today.